colossalai/tensor/_ops/linear.py

```diff
@@ -19,12 +20,31 @@ def colo_linear(types, args, kwargs, pg):
             bias = None
     else:
         bias = kwargs.get('bias', None)
-    
+
     if isinstance(bias, ColoTensor):
         bias = bias.torch_tensor()
 
     # Add communication logic before and after linear call.
     if isinstance(weight, ColoTensor):
-        return torch.nn.functional.linear(input_tensor, weight.torch_tensor(), bias)
+        if weight.shard_spec == None:
+            return torch.nn.functional.linear(input_tensor, weight.torch_tensor(), bias)
+        elif weight.shard_spec == '1Drow':
+            """
+            Input:S[1] x Weight:S[0] = Output:P
+            All-Reduce(Output) + bias = res
+            """
+            # Input:S[1]
+            input_per_partition = split_forward_gather_backward(input_tensor, ParallelMode.PARALLEL_1D, dim=-1)
```

Comment:
What if the input has already been split by the row linear?

Reply:
Do you mean the input has already been split by 1D-Col split?

Reply:
This pr is just working for complete input. The other situation will also be considered in the future.

Reply:
Ok, cool.

Reply:
we can check the input shape here. Make it works for our supported cases!
@Wesley-Jzy